config_hub/pretrain/tinyllama.yaml

```diff
@@ -1,29 +1,95 @@
-model_name: tiny-llama-1.1b
-logger_name: tensorboard
+
+# The name of the model to pretrain. Choose from names in ``litgpt.config``. Mutually exclusive with
+# ``model_config``. (type: Optional[str], default: null)
+model_name:
+
+# A ``litgpt.Config`` object to define the model architecture. Mutually exclusive with
+# ``model_config``. (type: Optional[Config], default: null)
+model_config:
+
+# Where to save checkpoints and logs. If running in a Lightning Studio Job, look for it in
+# /teamspace/jobs/<job-name>/share. (type: <class 'Path'>, default: out/pretrain)
+out_dir: out/pretrain
+
+# Path to a checkpoint directory to resume from in case training got interrupted. Or ``True`` to resume
+# from the latest checkpoint in ``out_dir``. (type: Union[bool, Path], default: False)
 resume: false
-devices: auto
-seed: 42
-data: TinyLlama
-out_dir: out/pretrain/tiny-llama
-tokenizer_dir: checkpoints/meta-llama/Llama-2-7b-hf
+
+# Data related arguments. If not provided, the default is ``litgpt.data.TinyLlama``. (type: Optional[LitDataModule], default: null, known subclasses: litgpt.data.LitDataModule, litgpt.data.Alpaca, litgpt.data.Alpaca2k, litgpt.data.AlpacaGPT4, litgpt.data.Dolly, litgpt.data.JSON, litgpt.data.Deita, litgpt.data.FLAN, litgpt.data.LIMA, litgpt.data.LongForm, litgpt.data.TinyLlama, litgpt.data.TinyStories, litgpt.data.OpenWebText)
```

Comment:
this is probably too much, hmm

Reply:
For pretraining, #1084 would help. But in general I don't know a solution to this. We could probably write a custom config serializer but I wouldn't suggest it